src/init.cpp

```diff
@@ -381,7 +381,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-externalip=<ip>", _("Specify your own public address"));
     strUsage += HelpMessageOpt("-forcednsseed", strprintf(_("Always query for peer addresses via DNS lookup (default: %u)"), DEFAULT_FORCEDNSSEED));
     strUsage += HelpMessageOpt("-listen", _("Accept connections from outside (default: 1 if no -proxy or -connect)"));
-    strUsage += HelpMessageOpt("-listenonion", strprintf(_("Automatically create Tor hidden service (default: %d)"), DEFAULT_LISTEN_ONION));
+    strUsage += HelpMessageOpt("-listenonion", strprintf(_("Run Tor hidden service (0 - disabled, 1 - normal mode, 2 - single onion service, default: %d)"), DEFAULT_LISTEN_ONION));
```

Comment:
I'm assuming the problem will be with %d and int64_t. I see you've changed DEFAULT_LISTEN_ONION from a bool to int64_t but did not change it here.

The format for int64_t (to (str)printf etc) is typically
- "I64d" (Win32/Win64)
-  "lld" (32-bit platforms) 
- "ld" (64-bit platforms) 

I suspect int64_t might not be the right type to use for DEFAULT_LISTEN_ONION.